cxx/util_io.cc

```diff
@@ -36,22 +36,33 @@ T_schema load_schema(const std::string& path) {
   return schema;
 }
 
-T_observations load_observations(const std::string& path) {
+ObservationVariant observation_string_to_value(const std::string& value_str, 
+                                               const std::string& distribution) {
+  if (distribution == "normal" || distribution == "bernoulli") {
+    return std::stod(value_str);
+  } else {
```

Comment:
I think it would be better to check here for bigram and explicitly fail if an unknown distribution name is passed.

Reply:
Done.